itwm_example/mco/space_sampling/tests/test_space_samplers.py

```diff
@@ -0,0 +1,178 @@
+                [0.75, 0.25, 0.0],
+                [0.75, 0.0, 0.25],
+                [0.50, 0.50, 0.0],
+                [0.50, 0.25, 0.25],
+                [0.50, 0.0, 0.50],
+                [0.25, 0.75, 0.0],
+                [0.25, 0.50, 0.25],
+                [0.25, 0.25, 0.50],
+                [0.25, 0.0, 0.75],
+                [0.0, 1.0, 0.0],
+                [0.0, 0.75, 0.25],
+                [0.0, 0.50, 0.50],
+                [0.0, 0.25, 0.75],
+                [0.0, 0.0, 1.0],
+            ],
+            list(self.generate_values(3, 5, with_zero_values=True)),
+        )
+
+        self.assertEqual(
+            [[1.0]], list(self.generate_values(1, 5, with_zero_values=True))
+        )
+        self.assertEqual(
+            [[1.0, 0.0], [0.75, 0.25], [0.50, 0.50], [0.25, 0.75], [0.0, 1.0]],
+            list(self.generate_values(2, 5, with_zero_values=True)),
+        )
+
+        self.assertEqual(
+            [
+                [1.0, 0.0, 0.0],
+                [0.875, 0.125, 0.0],
+                [0.875, 0.0, 0.125],
+                [0.75, 0.25, 0.0],
+                [0.75, 0.125, 0.125],
+                [0.75, 0.0, 0.25],
+                [0.625, 0.375, 0.0],
+                [0.625, 0.25, 0.125],
+                [0.625, 0.125, 0.25],
+                [0.625, 0.0, 0.375],
+                [0.5, 0.5, 0.0],
+                [0.5, 0.375, 0.125],
+                [0.5, 0.25, 0.25],
+                [0.5, 0.125, 0.375],
+                [0.5, 0.0, 0.5],
+                [0.375, 0.625, 0.0],
+                [0.375, 0.5, 0.125],
+                [0.375, 0.375, 0.25],
+                [0.375, 0.25, 0.375],
+                [0.375, 0.125, 0.5],
+                [0.375, 0.0, 0.625],
+                [0.25, 0.75, 0.0],
+                [0.25, 0.625, 0.125],
+                [0.25, 0.5, 0.25],
+                [0.25, 0.375, 0.375],
+                [0.25, 0.25, 0.5],
+                [0.25, 0.125, 0.625],
+                [0.25, 0.0, 0.75],
+                [0.125, 0.875, 0.0],
+                [0.125, 0.75, 0.125],
+                [0.125, 0.625, 0.25],
+                [0.125, 0.5, 0.375],
+                [0.125, 0.375, 0.5],
+                [0.125, 0.25, 0.625],
+                [0.125, 0.125, 0.75],
+                [0.125, 0.0, 0.875],
+                [0.0, 1.0, 0.0],
+                [0.0, 0.875, 0.125],
+                [0.0, 0.75, 0.25],
+                [0.0, 0.625, 0.375],
+                [0.0, 0.5, 0.5],
+                [0.0, 0.375, 0.625],
+                [0.0, 0.25, 0.75],
+                [0.0, 0.125, 0.875],
+                [0.0, 0.0, 1.0],
+            ],
+            list(self.generate_values(3, 9, with_zero_values=True)),
+        )
+
+        self.assertEqual(
+            [[0.75, 0.25], [0.50, 0.50], [0.25, 0.75]],
+            list(self.generate_values(2, 5, with_zero_values=False)),
+        )
+
+        self.assertEqual(
+            list(self.generate_values(3, 9, with_zero_values=False)),
```

Comment:
Nitpick - expected value order

Reply:
True. Also went through the rest of the itwm code base, fixing similar things.